staging/src/k8s.io/kubectl/pkg/cmd/config/unset.go

```diff
@@ -104,6 +135,14 @@ func (o *unsetOptions) complete(cmd *cobra.Command, args []string) error {
 	}
 
 	o.propertyName = args[0]
+
+	// try to determine if we have a jsonpath from first character of first argument
+	// this should only ever be a { if this is a jsonpath string, if it is not we will try to use the old dot delimited
+	// syntax instead.
```

Comment:
I am curious why the old implementation is left rather than being removed since jsonpath appears to be a superset of the old style. 

It seems to me you could turn any of the old queries into a json path query simply by prefixing it with "$.". Is that true?

Reply:
Unfortunately it is not possible to simply translate the old syntax to the new syntax. It is definitely possible, however it seems like too much extra effort for something we intend to deprecate eventually anyway.

As an example, a translation of setting a cluster resource's server field would be:

Original - `clusters.test-cluster.server`
jsonpath - `{.clusters[?(@.name=="test-cluster")].cluster.server`

The only way to translate these would be to hard code the fact that the middle step in the path (`test-cluster`) would need to be dropped into the hard coded `[?(@.name=="$")].cluster`, which, while doable, implements some pretty ugly code that we will need to remove later anyway. If we wanted to do this without hardcoding values (which is the only way I'd be willing to do it) we would need to create another set of logic to basically search for the fields that the associate clientcmd types use, the associated clientcmdapiv1 types use, then do a lookup to figure out what field we're doing the "filter" on, which is where the name in `@.name` comes from, and then find the field we need to put in the next node in the jsonpath, which can't really be "gotten" in a controller way and would have to be hard coded to cluster anyway. At best we would split and look backwards at clusters and parse out the singular form of the word which is still pretty icky imo.

tldr jsonpath isn't really a super set but more of a parallel of the old style that does overlap in several areas but unfortunately doesn't allow for a bi-directional mapping between the two.